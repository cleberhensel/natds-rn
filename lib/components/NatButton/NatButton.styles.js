import { Platform } from 'react-native';
import styled, { css } from 'styled-components/native';
import { roboto, sanfrancisco } from '../../assets/theme/fonts/fonts';
import { colors } from '../../assets/theme/colors/colors';

const buttonStyles = {
  card: {
    backgroundColor: colors.primary.yellow,
    color: colors.primary.white,
    height: 128,
    width: 128,
  },
  default: {
    backgroundColor: colors.primary.yellow,
    color: colors.primary.white,
    height: 48,
  },
  disabled: {
    backgroundColor: colors.bw.grayLight,
    color: colors.bw.grayDark,
    height: 48,
  },
  link: {
    backgroundColor: 'transparent',
    color: colors.content.info,
    height: 48,
  },
  newDefault: {
    backgroundColor: colors.primary.yellow,
    color: colors.bw.grayDarker,
    height: 48,
  },
  newShort: {
    backgroundColor: colors.primary.white,
    color: colors.bw.grayDarker,
    height: 40,
    width: 328,
  },
  outline: {
    backgroundColor: colors.primary.white,
    borderColor: colors.primary.yellow,
    borderWidth: 2,
    color: colors.primary.yellow,
    height: 48,
  },
  short: {
    backgroundColor: colors.primary.white,
    color: colors.primary.yellow,
    height: 40,
    width: 328,
  },
};

const getStylesFrom = styles => type => styles[type] || styles.default;

const getButtonStyles = getStylesFrom(buttonStyles);

export const ButtonWrapper = styled.TouchableOpacity`
  background-color: ${({ type }) => getButtonStyles(type).backgroundColor};
  height: ${({ type }) => getButtonStyles(type).height};
  width: ${({ type }) => getButtonStyles(type).width || '100%'};
  border-radius: 6;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  border-color: ${({ type }) => getButtonStyles(type).borderColor || 'transparent'};
  border-width: ${({ type }) => getButtonStyles(type).borderWidth || 0};
`;

export const ButtonText = styled.Text`
  color: ${({ type }) => getButtonStyles(type).color};
  font-size: 14px;
  align-self: center;
  font-weight: bold;
  letter-spacing: 1;
  font-family:  ${(Platform.OS === 'ios') ? sanfrancisco.bold : roboto.bold};

  ${props => props.type === 'card' && css`
    width: 64px;
    text-align: center;
    line-height: 21px;
  `}
`;

export const ButtonIcon = styled.Image`
  height: 30;
  width: 30;
`;
