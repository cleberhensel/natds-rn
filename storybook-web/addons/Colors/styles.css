.colors {
  font-weight: 400;
  line-height: 1.5;
}

.colors__container {
  max-width: 1400px;
  display: flex;
  flex-flow: row wrap;
  padding: 60px;
}

.colors__group {
  display: flex;
  flex-flow: row wrap;
  flex: 0 0 100%;
  margin-bottom: 30px;
}

.colors__group__title {
  font-size: 1.5rem;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-weight: 400;
  color: #555;
  flex: 0 0 100%;
  padding: 0 0 20px;
  text-transform: capitalize;
  margin: 0;
}

.colors__group__list {
  flex: 0 0 100%;
  display: flex;
  flex-flow: row wrap;
  padding: 0;
  margin: 0;
}

.colors__item {
  width: 130px;
  display: flex;
  flex: 0 0 auto;
  position: relative;
  flex-flow: column;
  overflow: hidden;
  padding: 0;
  margin: 0 15px 15px 0;
}

.colors__item__color {
  width: 130px;
  height: 130px;
  margin-bottom: 5px;
}

.colors__item__color-name,
.colors__item__name {
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-size: 0.875rem;
  line-height: 1rem;
  padding: 0;
  margin: 0;
}

.colors__item__name {
  color: #888;
  margin-bottom: 5px;
}

.colors__item__color-name {
  color: #555;
}
