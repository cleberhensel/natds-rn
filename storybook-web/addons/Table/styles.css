.table__container {
  font-weight: 400;
  line-height: 1.5;
  max-width: 800px;
  width: 100%;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 60px;
}

.table__title {
  font-size: 18px;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 0 20px;
  margin: 0;
}

.table {
  width: 100%;
  border-collapse: collapse;
}

.table__body__row {
  background: #f9f9f9;
  border-bottom: #fff 1px solid;
  padding: 10px;
}

.table__head__item:first-child {
  width: 25%;
}

.table__head__item,
.table__body__item {
  font-size: 0.875rem;
  text-align: left;
}

.table__head__item {
  padding-bottom: 10px;
  text-transform: capitalize;
}

.table__body__item {
  line-height: 1.25rem;
  font-weight: 400;
  height: 20px;
  padding: 10px;
}

.table__body__item__color {
  border: #ddd 2px solid;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  display: inline-block;
  vertical-align: top;
  margin: 2px 10px 0;
}

.table__body__item__color-name {
  line-height: 1rem;
  display: inline-block;
}
