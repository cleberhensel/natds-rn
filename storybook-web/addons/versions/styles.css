.versions {
  font-weight: 400;
  line-height: 1.5;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 30px;
}

.versions__type {
  padding: 0 0 10px;
  margin: 0 10px 10px;
}

.versions__type__name {
  border-bottom: rgba(0,0,0,.1) 1px solid;
  font-size: 18px;
  text-transform: capitalize;
  padding: 0 0 10px;
  margin: 0 0 10px;
}

.versions__type__warning {
  font-size: 18px;
  text-transform: capitalize;
  margin-bottom: 10px;
}

.versions__type__button {
  background: rgba(0,0,0,.3);
  border: 0;
  border-radius: 4px;
  font-size: 1rem;
  color: #fff;
  text-transform: capitalize;
  cursor: pointer;
  padding: 7px 14px;
  margin: 0 10px 10px 0;
}

.versions__type__button:hover {
  background: rgba(0,0,0,.2);
}

.versions__type__button:active {
  background: rgba(0,0,0,.25);
}

.versions__type__button--selected,
.versions__type__button--selected:hover {
  background: rgba(0,0,0,.6);
  cursor: default;
}
