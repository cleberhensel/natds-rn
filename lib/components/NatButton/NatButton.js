import React from 'react';
import PropTypes from 'prop-types';
import icoArrowLeft from '../../assets/icons/ic_arrow_left_white.png';

import { ButtonWrapper, ButtonText, ButtonIcon } from './NatButton.styles';

export const NatButton = ({
  onPress, type, label, noUpperCase, iconLeft,
}) => (
    <ButtonWrapper
      accessible
      accessibilityRole="button"
      onPress={onPress}
      type={type}
      disabled={type === 'disabled'}
    >
      {iconLeft && <ButtonIcon source={iconLeft} />}
      <ButtonText
        type={type}
        accessibilityLabel={`${label}`}
      >
        {noUpperCase ? label : label.toUpperCase()}
      </ButtonText>
      {type === 'card' && <ButtonIcon source={icoArrowLeft} />}
    </ButtonWrapper>
);

NatButton.propTypes = {
  iconLeft: PropTypes.string,
  label: PropTypes.string.isRequired,
  noUpperCase: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
  type: PropTypes.string,
};

NatButton.defaultProps = { type: 'default' };
