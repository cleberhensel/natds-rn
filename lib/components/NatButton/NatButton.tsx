import React from "react";
import icoArrowLeft from "../../assets/icons/ic_arrow_left_white.png";
import { ButtonWrapper, ButtonText, ButtonIcon } from "./NatButton.styles";
type NatButtonProps = {
  iconLeft?: string,
  label: string,
  noUpperCase?: boolean,
  onPress: (...args: any[]) => any,
  type?: string
};
export const NatButton: React.SFC<NatButtonProps> = ({
  onPress,
  type,
  label,
  noUpperCase,
  iconLeft
}) => (
  <ButtonWrapper
    accessible
    accessibilityRole="button"
    onPress={onPress}
    type={type}
    disabled={type === "disabled"}
  >
    {iconLeft && <ButtonIcon source={iconLeft} />}
    <ButtonText type={type} accessibilityLabel={`${label}`}>
      {noUpperCase ? label : label.toUpperCase()}
    </ButtonText>
    {type === "card" && <ButtonIcon source={icoArrowLeft} />}
  </ButtonWrapper>
);
NatButton.defaultProps = {
  type: "default"
};
