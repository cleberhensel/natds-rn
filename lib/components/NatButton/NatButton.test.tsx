import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-styled-components/native";
import { Text } from "react-native";
import { NatButton } from "./NatButton";
import { ButtonIcon } from "./NatButton.styles";
describe("Button component default", () => {
  it("Should render button component ", () => {
    const button = renderer
      .create(
        <NatButton label="string" onPress={jest.fn()}>
          <Text>Oi</Text>
        </NatButton>
      )
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  describe("when is card type", () => {
    let props = {};
    beforeEach(() => {
      props = {
        label: "test",
        onPress: jest.fn(),
        type: "card"
      };
    });
    it("Should render button component ", () => {
      const button = renderer.create(<NatButton {...props} />).toJSON();
      expect(button).toMatchSnapshot();
    });
    it("should have an icon", () => {
      const wrapper = shallow(<NatButton {...props} />);
      expect(wrapper.find(ButtonIcon).length).toBe(1);
    });
  });
  describe("when is short type", () => {
    let props = {};
    beforeEach(() => {
      props = {
        label: "test",
        onPress: jest.fn(),
        type: "short"
      };
    });
    it("Should render short button component", () => {
      const button = renderer.create(<NatButton {...props} />).toJSON();
      expect(button).toMatchSnapshot();
    });
    it("should not have an icon", () => {
      const wrapper = shallow(<NatButton {...props} />);
      expect(wrapper.find(ButtonIcon).length).toBe(0);
    });
  });
  describe("when is outline type", () => {
    let props = {};
    beforeEach(() => {
      props = {
        label: "test",
        onPress: jest.fn(),
        type: "outline"
      };
    });
    it("Should render outline button component", () => {
      const button = renderer.create(<NatButton {...props} />).toJSON();
      expect(button).toMatchSnapshot();
    });
    it("should not have an icon", () => {
      const wrapper = shallow(<NatButton {...props} />);
      expect(wrapper.find(ButtonIcon).length).toBe(0);
    });
  });
});
