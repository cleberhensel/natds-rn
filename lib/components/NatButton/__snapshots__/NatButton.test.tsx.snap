// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component default Should render button component  1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4AB34",
      "borderColor": "transparent",
      "borderRadius": 6,
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    accessibilityLabel="string"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": 1,
        },
      ]
    }
    type="default"
  >
    STRING
  </Text>
</View>
`;

exports[`Button component default when is card type Should render button component  1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4AB34",
      "borderColor": "transparent",
      "borderRadius": 6,
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 128,
      "justifyContent": "center",
      "opacity": 1,
      "width": 128,
    }
  }
>
  <Text
    accessibilityLabel="test"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": 1,
          "lineHeight": 21,
          "textAlign": "center",
          "width": 64,
        },
      ]
    }
    type="card"
  >
    TEST
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../lib/assets/icons/ic_arrow_left_white.png",
      }
    }
    style={
      Array [
        Object {
          "height": 30,
          "width": 30,
        },
      ]
    }
  />
</View>
`;

exports[`Button component default when is outline type Should render outline button component 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#F4AB34",
      "borderRadius": 6,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    accessibilityLabel="test"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#F4AB34",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": 1,
        },
      ]
    }
    type="outline"
  >
    TEST
  </Text>
</View>
`;

exports[`Button component default when is short type Should render short button component 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "transparent",
      "borderRadius": 6,
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 328,
    }
  }
>
  <Text
    accessibilityLabel="test"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#F4AB34",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": 1,
        },
      ]
    }
    type="short"
  >
    TEST
  </Text>
</View>
`;
